File: lab7/Java/src/Lab7.java
import enums.LogLevel;
import java.io.IOException;
import logger.*;

public class Lab7 {
    public static void main(String[] args) throws IOException {
        Logger logger = new Logger();
        logger.RegisterLogger(new FileLogger("logs.txt"));
        logger.RegisterLogger(new ConsoleLogger());
        logger.Log("Демонстраци мульти-логгера", LogLevel.INFO);
        logger.RemoveLogger(logger.GetWrappers().get(1).getId());
        logger.Log("Сообщение после удаления коснольного логгера", LogLevel.WARN);
    }
}

File: lab7/C++/src/lab7.cpp
#include <iostream>
#include <memory>
#include "Logger.hpp"
#include "FileLogger.hpp"
#include "ConsoleLogger.hpp"

using namespace std;

int main()
{
    setlocale(LC_ALL, "Rus");
    auto logger = make_unique<Logger>();
    logger->RegisterLogger(make_unique<FileLogger>("logs.txt"));
    logger->RegisterLogger(make_unique<ConsoleLogger>());
    logger->Log("Демонстрация мульти-логгера", LogLevel::eInfo);
    logger->RemoveLogger(logger->GetLoggers()[1]->GetId());
    logger->Log("Сообщение после удаления консольного логгера", LogLevel::eWarn);
}
